src/mmw/js/src/modeling/models.js

```diff
@@ -207,8 +252,73 @@ var ScenarioModel = Backbone.Model.extend({
         // model, we shouldn't reset them from the server. Pull them off of
         // the response to prevent overwriting them.
         delete response.modifications;
+
+        // TODO We don't want to set the results until a future
+        // PR when we intentionally cache results.
+        delete response.results;
+
         return response;
-     }
+    },
+
+    // Poll the taskModel for results and reset the results collection when done.
+    // If not successful, the results collection is reset to be empty.
+    pollForResults: function() {
+        var self = this,
+            taskModel = this.get('taskModel'),
+            setPolling = function(polling) {
+                self.get('results').forEach(function(resultModel) {
+                    resultModel.set('polling', polling);
+                });
+            },
+            setNullResults = function() {
+                self.get('results').forEach(function(resultModel) {
+                    resultModel.set('result', null);
+                });
+            },
+            setResults = function() {
+                var rawServerResults = taskModel.get('result');
+                if (rawServerResults == "" || rawServerResults == null) {
+                    setNullResults();
+                } else {
+                    var serverResults = JSON.parse(rawServerResults);
+                    self.get('results').forEach(function(resultModel) {
+                        resultModel.set('result', serverResults[resultModel.get('name')]);
+                    });
+                }
```

Comment:
Instead of micromanaging results client-side, can we just populate the entire results collection with whatever the server responds with?